code/game/objects/items/melee/misc.dm

```diff
@@ -464,6 +467,11 @@
 	if(!on)
 		return ..()
 
+	if(!owner_data || owner_data?.owner?.current != user)
+		return ..()
+
+	var/is_target = owner_data.contractor_hub?.current_contract?.contract?.target == target.mind
```

Comment:
This is only used once, you don't need to create var for it

Reply:
Looks nicer than a super long if line